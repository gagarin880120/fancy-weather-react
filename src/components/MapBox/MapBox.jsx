import React, { useRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import mapboxgl from 'mapbox-gl';
import styles from './MapBox.module.css';

export default function MapBox({ lon, lat, mapZoom, onMapMove }) {
  const mapContainer = useRef();
  mapboxgl.accessToken = 'pk.eyJ1IjoiZ2FnYXJpbjg4MDEyMCIsImEiOiJjazNicHc1NW4wOHFvM251aXg3YzlheDFmIn0.C2XPJXYAzS1e0OjVNFC-NQ';
  useEffect(() => {
    const map = new mapboxgl.Map({
      container: mapContainer.current,
      style: 'mapbox://styles/mapbox/streets-v11',
      center: [lon, lat],
      zoom: mapZoom,
    });
    map.on('move', () => {
      onMapMove(map.getZoom().toFixed(2));
    });
  }, [mapContainer, lon, lat]);
  return (
    <div ref={mapContainer} className={styles.container} />
  );
}

MapBox.propTypes = {
  lon: PropTypes.number,
  lat: PropTypes.number,
  mapZoom: PropTypes.string,
  onMapMove: PropTypes.func,
};

MapBox.defaultProps = {
  lon: 0,
  lat: 0,
  mapZoom: '8',
  onMapMove: null,
};
