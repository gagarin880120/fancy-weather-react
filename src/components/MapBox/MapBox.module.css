.container {
  width: 400px;
  height: 400px;
  position: absolute;
  right: 5%;
  top: 10%;
  border-radius: 50%;
}
