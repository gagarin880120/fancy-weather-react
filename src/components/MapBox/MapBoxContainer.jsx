import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MapBox from './MapBox';
import { selectLatitude, selectLongitude } from '../../redux/selectors';

export function MapBoxContainer({ lon, lat }) {
  return (
    <MapBox
      lon={lon}
      lat={lat}
    />
  );
}

export const mapStateToProps = (state) => ({
  lon: selectLongitude(state),
  lat: selectLatitude(state),
});

MapBoxContainer.propTypes = {
  lon: PropTypes.number,
  lat: PropTypes.number,
};

MapBoxContainer.defaultProps = {
  lon: 0,
  lat: 0,
};

export default connect(mapStateToProps)(MapBoxContainer);
