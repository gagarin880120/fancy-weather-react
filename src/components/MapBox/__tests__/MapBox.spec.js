import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import MapBox from '../MapBox';
import mapboxgl from 'mapbox-gl';

jest.mock('mapbox-gl', () => ({
  Map: () => ({
    getZoom: () => ({
      toFixed: jest.fn(),
    }),
    on: jest.fn((type, listener = jest.fn()) => {
      listener();
    }),
  }),
  Marker: () => ({
    setLngLat: jest.fn(),
    addTo: jest.fn(),
  }),
}));

describe('MapBox', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<MapBox />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });

  describe('mount with testRenderer', () => {
    const onMapMove = jest.fn();
    const testRenderer = TestRenderer.create(
    <MapBox
      lon={23}
      lat={52}
      onMapMove={onMapMove}
    />
    );
    const map = new mapboxgl.Map();
    test('when latitude and longitude are updated', () => {
      act(() => {
        testRenderer.update(
          <MapBox
            lon={25}
            lat={55}
            onMapMove={onMapMove}
            zoom="8"
          />
        );
      });

      act(() => {
        map.on();
      })
      expect(onMapMove).toHaveBeenCalled();
    });
  });
});
