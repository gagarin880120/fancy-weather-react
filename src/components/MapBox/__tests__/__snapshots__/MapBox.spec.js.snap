// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapBox component should render without crashing 1`] = `
<div
  className="container"
/>
`;
